### DISCUSSION TITLE
Firmware Flashing - Noob Basics

### DISCUSSION THREAD

**Discussion by doubledipped** [shanteacontrols/OpenDeck#193] [Q&A]
I realise that this is probably a very basic question, but I am struggling to find an answer anywhere.

I received an OpenDeck M board a couple of years ago as a gift and I have only just finished building it into a custom controller for my MIDI needs. This is the first project of this kind I have undertaken, so there was a bit of a learning curve for me, learning how solder & crimp connections, sourcing components etc.

I would like to flash the firmware on the board, which is currently running 5.x. Unfortunately, when I try to use the configurator (https://config.shanteacontrols.com/#/) to put the board into software bootloader mode, the configurator simply reports that no OpenDeck board is present, so I can't use the firmware update feature.

I guess that means that I have to use the hardware flashing method and that is where I am stuck. The instructions require jumping a couple of headers to ensure that the boards starts in bootloader mode, but how do I connect to it? I have used  UART connections on JTAG interfaces to install firmware on routers and access points etc. but I don't believe that is the case here. Should I be able to connect to the board using the USB interface? Should the USB interface appear as a COM port on my Windows 10 PC? I have the Windows build of 'avrdude' ready to copy the .sysex file I downloaded, but I still don't understand how the physical connectivity is established. 

Like I say, basic stuff, but any pointers gratefully received.

**Comment by paradajz** [comment]
I assume that is the v2.x.x board, right? If it's running v5.x.x firmware then in must be, since the older variants of the M board used AVR MCU instead of the STM32F4 one and the latest firmware for those boards is v4.x.x. Not quite sure why the board wouldn't work in bootloader mode, but at this point, I'm not even sure that upgrading from v5 to the latest firmware would even work through the OpenDeck bootloader. So what you need to do is to find SWDIO/SWDCLK pin header on the board and connect ST Link to it or some other programmer and the load the latest .hex file present in GitHub release section. Also, ground connection is required between the board and the programmer - you can use any GND pin from analog headers from that (it was a mistake not to include it in SWD header).

**Reply by doubledipped** [reply]
Thank you for getting back to me so quickly. Yes - sorry I forgot to mention that it is a v2 board. Also, I did wonder whether the UI method would support the older v5 firmware or even whether upgrading across so many versions would work. Thank you very much for the guidance on how to connect to the board and the file type I need to load, that's all I need to get me back on track as I was floundering. Thank you again @paradajz, I really appreciate your help and absolutely love OpenDeck.
